src/bun.js/bindings/NodeVM.cpp

```diff
@@ -200,6 +206,72 @@ JSC::JSFunction* constructAnonymousFunction(JSC::JSGlobalObject* globalObject, c
     return function;
 }
 
+JSInternalPromise* importModule(JSGlobalObject* globalObject, JSString* moduleNameValue, JSValue parameters, const SourceOrigin& sourceOrigin)
+{
+    if (auto* fetcher = sourceOrigin.fetcher(); !fetcher || fetcher->fetcherType() != ScriptFetcher::Type::NodeVM) {
+        return nullptr;
+    }
+
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto* fetcher = static_cast<NodeVMScriptFetcher*>(sourceOrigin.fetcher());
+
+    JSValue dynamicImportCallback = fetcher->dynamicImportCallback();
+
+    if (!dynamicImportCallback || !dynamicImportCallback.isCallable()) {
+        return nullptr;
+    }
+
+    JSFunction* owner = fetcher->owner();
+
+    MarkedArgumentBuffer args;
+    args.append(moduleNameValue);
+    args.append(owner ? owner : jsUndefined());
+    args.append(parameters);
+
+    RETURN_IF_EXCEPTION(scope, nullptr);
+
+    JSValue result = AsyncContextFrame::call(globalObject, dynamicImportCallback, jsUndefined(), args);
+
+    RETURN_IF_EXCEPTION(scope, nullptr);
+
+    if (auto* promise = jsDynamicCast<JSInternalPromise*>(result)) {
+        return promise;
+    }
+
+    auto* promise = JSInternalPromise::create(vm, globalObject->internalPromiseStructure());
+
+    RETURN_IF_EXCEPTION(scope, nullptr);
+
+    auto* transformer = JSNativeStdFunction::create(vm, globalObject, 1, {}, [](JSGlobalObject* globalObject, CallFrame* callFrame) -> JSC::EncodedJSValue {
+        VM& vm = globalObject->vm();
+        auto scope = DECLARE_THROW_SCOPE(vm);
+
+        JSValue argument = callFrame->argument(0);
+
+        if (JSObject* object = argument.getObject()) {
+            JSValue result = object->get(globalObject, JSC::Identifier::fromString(vm, "namespace"_s));
+            RETURN_IF_EXCEPTION(scope, {});
+            if (result && !result.isUndefinedOrNull()) {
```

Comment:
```suggestion
            if (!result.isUndefinedOrNull()) {
```
`get` won't return an empty value if there isn't an exception